src/Integration/Notifications/VisualStudioStatusBarHelper.cs

```diff
@@ -59,17 +56,22 @@ public static void AddStatusBarIcon(FrameworkElement statusBarIcon)
             dockPanel.Children.Insert(sourceControlHostIndex + 1, statusBarIcon);
         }
 
-        private static DockPanel GetStatusBarPanel(DependencyObject mainWindow)
+        private static DockPanel FindSccStatusBarParentPanel(Window window)
         {
-            for (int i = 0; i < VisualTreeHelper.GetChildrenCount(mainWindow); i++)
-            {
-                var dockPanel = VisualTreeHelper.GetChild(mainWindow, i) as DockPanel;
-                if (dockPanel?.Name == StatusBarPanelName)
-                {
-                    return dockPanel;
-                }
-            }
-            return null;
+            // We want to add our icon as in the VS status bar as a sibling of the source control visuals.
+
+            // Assumptions:
+            // 1) there is a control named "PART_SccStatusBarHost"
+            // 2) that control is hosted in a dock panel 
+
+            // This approach is a bit ugly, but it's also the approach used in the extension -
+            // see https://github.com/github/VisualStudio/blob/824bab4ab2c3d8f6787cf38ff5ff9d4e9df00e98/src/GitHub.InlineReviews/Services/PullRequestStatusBarManager.cs#L171
+
+            // Note: the visual tree of VS2019 is different from that of VS2015 and VS2017, but they
+            // all have contain this named PART control. See #1751.
+
+            var statusBarPart = window?.Template?.FindName(SccStatusBarHostName,  window) as FrameworkElement;
+            return statusBarPart?.Parent as DockPanel;
         }
 
         private static int GetStatusBarSourceControlHostIndex(DockPanel dockPanel)
```

Comment:
This code is a bit odd.

FindSccStatusBarParentPanel is finding a framework element named SccStatusBarHostName.
We then take that element's parent and pass it to GetStatusBarSourceControlHostIndex, which searches for SccStatusBarHostName again, to get its index.

It's a bit too complicated and I think it can be simplified. I've created a branch with a proposal fix: https://github.com/SonarSource/sonarlint-visualstudio/pull/2167/files#